### DISCUSSION TITLE
alarm 8 while auto-squaring Y axis

### DISCUSSION THREAD

**Discussion by d-granger** [phil-barrett/grblHAL-teensy-4.x#96] [General]
I have a CNC gantry router with dual Y axes.  I used the grblHAL web builder to configure my kernel with 3 axes, auto-squared.  I have my limits placed in the same fashion as in Phil's blog on auto-squaring (min and max on Y and min only on A).  I'm using 24v NO NPN prox sensors for limits so I have an 8 chan opto-isolated level shifting board between the sensors and the limit inputs to the BOB.  The second Y axis limit is connected to the A limit on the BOB.  I inverted the limits for XYZ in ioSender.  Hard limits are enabled as is homing via ioSender.

Manually tripping the limits shows that A is indeed coupled to Y as ioSender shows Y as tripped when I trigger A.  My level-shifting board has indicator leds on each channel and shows that the signal is coming in on the A limit.

When I home, Z is successful as is X, then the Y axis moves to the minimum position until both Y & A limits trip (leds on both sensors and both inputs on opto board light), and ioSender raises an Alarm 8 & halts leaving both limits tripped.  I adjusted my limit stops so Y tripped before A and vice versa, and the behavior repeats in both cases.

Any idea on how to remedy this?

**Comment by phil-barrett** [comment]
Increase pull off distance? I think the default is 1mm. It needs to be large enough to untrigger the sensor. I'd start with 5mm.

**Reply by d-granger** [reply]
Thank you!

The magic distance was 1.5mm.  I obviously haven't calibrated my squaring and hope that once I have it dialed in I won't be needing a large pull off.  I was careful during construction to square the frame but I'm sure there will still be some offset between the 2 rails.  Thanks again.

**Reply by phil-barrett** [reply]
1.5mm is not bad at all.  Also, if you increase the search seek rate ($25), you will get more overshoot and will have to increase the back off distance.

**Reply by d-granger** [reply]
I bumped my search seek to 1500mm/sec so I expect that there is some overshoot from that.  I still hope that my Y and A are pretty close since I was careful about getting things square during the mechanical build.
